javascript/javascript1/week4/voice.js

```diff
@@ -0,0 +1,123 @@
+function getReply(command) {
+  let result;
+  if (command.includes("hello my name is")) {
+    const userName = saveName(command);
+    result = `nice to meet you ${userName}`;
+  } else if (command.includes(str2)) {
+    if (!str1) {
+      result = `you have not tell your name yet`;
+    }
+    const yourName = saveName(str1);
+    result = `Your name is ${yourName}`;
+  } else if (command.includes("add")) {
+    const activity = addActivity(command);
+    result = `${activity} added to your todo`;
+  } else if (command.includes("remove")) {
+    result = removeActivity(str5);
+  } else if (command.includes("what is on my")) {
+    const length = `You have ${listArr.length} todos - ${listArr.join(" , ")}`;
+    result = length;
+  } else if (command.includes(str7)) {
+    result = new Date().toLocaleDateString("en-US", {
+      day: "numeric",
+      month: "long",
+      year: "numeric",
+    });
+  } else if (command.startsWith("what is")) {
+    return calcOperations(command);
+  } else if (command.includes(str9)) {
+    setTimeout(timer, 4000);
+    result = "Timer set for 4 seconds";
+  } else if (command.includes("how will")) {
+    result = guessAboutToday();
+  } else {
+    result = "error";
+  }
+  return result;
+}
+function saveName(string) {
+  const stringArray = string.split(" ");
+  const userName = stringArray[stringArray.length - 1];
+  return userName;
+}
+function addActivity(string) {
+  const startIndex = string.indexOf("add") + 4;
+  const endIndex = string.indexOf("to my");
+  const activity = string.substring(startIndex, endIndex).trim();
+  listArr.push(activity);
+  return activity;
+}
+
+function removeActivity(string) {
+  const startIndex = string.indexOf("remove") + 7;
+  const endIndex = string.indexOf("from my");
+  const activity = string.substring(startIndex, endIndex).trim();
+  const index = listArr.indexOf(activity);
+  listArr.splice(index, 1);
+  return `Removed ${activity} from your todo`;
```

Comment:
This will be announced even if there is no activity removed. The correct way would be to make an `if else` where you check if the activity is in the `listArr` and then remove it. If it's not in the list, then `return` something else, as `Sorry, I couldn't find ${activity} in your todo`.

Reply:
Thank you for the review. I have changed the code.

Reply:
Nice. Remember to push the commit.